pkg/operator/ceohelpers/machine_api.go

```diff
@@ -63,3 +105,61 @@ func (m *MachineAPI) IsFunctional() (bool, error) {
 
 	return false, nil
 }
+
+// IsEnabled checks if the Machine API is enabled.
+// As of today Machine API is enabled when:
+// * clusterVersionInformer is synced
+// * clusterVersion has "MachineAPI" as an enabled capability
+// This method might not be reliable under some configuration, thus it's important to also check whether it IsAvailable
+func (m *MachineAPI) IsEnabled() (bool, error) {
+	if !m.hasVersionInformerSyncedFn() {
+		return false, fmt.Errorf("ClusterVersionInformer is not yet synced")
+	}
+
+	clusterVersion, err := m.versionLister.Get("version")
+	if err != nil {
+		return false, err
+	}
+
+	// this is a special case introduced from 4.14 on, where MachineAPI can be optional with clusters installed using capabilities.baselineCapabilitySet=None
+	// on upgrades from prior versions of OpenShift the status should have MachineAPI listed as an EnabledCapabilities
+	if clusterVersion != nil && clusterVersion.Spec.Capabilities != nil && string(clusterVersion.Spec.Capabilities.BaselineCapabilitySet) == "None" {
+		machineAPIEnabled := false
+		for _, capability := range clusterVersion.Status.Capabilities.EnabledCapabilities {
+			if string(capability) == "MachineAPI" {
```

Comment:
Nit: I probably would use the constant from `configv1` rather than string matching, but 🤷